website/docs/newaction.md

```diff
@@ -14,11 +14,14 @@ newaction { description }
 | shortname   | A short summary for the help text, e.g. "Visual Studio 2013".                      |
 | description | A description of the action's result, e.g. "Generate Visual Studio 2013 project files". |
 | execute     | A function to be executed when the action is fired.                                |
-| os          | If the toolset targets a specific OS, the [identifier](system.md) for that OS. |
+| os          | Deprecated, use targetos instead. |
+| targetos    | If the toolset targets a specific OS, the [identifier](system.md) for that OS. |
 | valid_kinds | The list of [project kinds](kind.md) supported by the action. |
 | valid_languages | The list of [languages](language.md) supported by the action. |
 | valid_tools | The list of [tools](toolset.md) supported by the action. |
+| toolset | Default [tools](toolset.md). |
 | onStart     | A callback marking the start of action processing. |
+| onSolution | Deprecated, use onWorkspace instead. |
```

Comment:
@starkos would it be better to just remove this function instead of documenting it now? This change happened so long ago now.

Reply:
Notice that one of the module I fixed (premake-ninja) still uses it. I didn't yet upgrade it, but will do it soon.

Premake's code doesn't warn it as deprecated neither BTW. (contrary to some other fields).

Even if removed in code, I think keeping it somewhere as "removed"/"renamed" might help migrating. (maybe in dedicated section).

But as premake5 (no-alpha) has not been out, we might remove them indeed.